pkg/controllers/nodeclass/status/images.go

```diff
@@ -154,14 +155,15 @@ func (r *NodeImageReconciler) Reconcile(ctx context.Context, nodeClass *v1beta1.
 	if len(goalImages) == 0 {
 		nodeClass.Status.Images = nil
 		nodeClass.StatusConditions().SetFalse(v1beta1.ConditionTypeImagesReady, "ImagesNotFound", "ImageSelectors did not match any Images")
-		logger.Info("no node images")
+		logger.Info("no available node images")
 		return reconcile.Result{RequeueAfter: 5 * time.Minute}, nil
 	}
 
+	if utils.HasChanged(nodeClass.Status.Images, goalImages, &hashstructure.HashOptions{SlicesAsSets: false}) {
```

Comment:
minor: Possibly worth a comment explaining why you set SlicesAsSets here to false?

```
// Not treating these slices as sets because we care about order - it impacts which one will eventually be chosen
```

or something like that?